tensorflow/core/grappler/optimizers/auto_mixed_precision.cc

```diff
@@ -1203,16 +1203,39 @@ Status AutoMixedPrecisionImpl::Optimize() {
   VLOG(2) << "Building node type map for graph";
   TF_RETURN_IF_ERROR(node_type_map_.Init(*graph_));
 
+  // Note: If an op is added to this list, it should also be added to the
+  // AddDataStructureOpsToMap call below (and to the clearlist if it involves
+  // data flow).
   // TODO(benbarsdell): Add support for TensorListPushBackBatch and
   // TensorListConcatLists. They require special handling because they connect
   // multiple list objects together. Currently if they appear in the graph then
   // we have no choice but to disallow changing any tensor list ops, as
   // otherwise we risk breaking the graph if some are changed and some are not
   // (within a connected cluster of tensor list nodes).
+  const gtl::FlatSet<string> supported_list_ops = {
+      "EmptyTensorList",
+      "TensorListSplit",
+      "TensorListFromTensor",
+      "TensorListReserve",
+      "TensorListScatter",
+      "TensorListScatterV2",
+      "TensorListPushBack",
+      "TensorListSetItem",
+      "TensorListScatterIntoExistingList",
+      "TensorListPopBack",
+      "TensorListStack",
+      "TensorListConcat",
+      "TensorListConcatV2",
+      "TensorListGetItem",
+      "TensorListGather"};
```

Comment:
TensorListLength and TensorListElementShape can be added to this list. And you should change the sentence "If an op is added to this list, it should also be added..." to "If an op is added to this list, it should **probably** also be added...".

Also maybe handle TensorListResize?

Reply:
Good point.

Does TensorListResize require special handling?

Reply:
You can treat TensorListResize as a read or write op, and everything will do the right thing. It isn't really either, but you need the `ForceColorMatchBetweenDataStructureOps` function to change the output dtype if it changes the input dtype, or vice versa.

You can also just ignore TensorListResize for now and fix it later.

Reply:
TensorListResize has no dtype attribute, so it can't be passed to AddDataStructureOpsToMap.

I may be missing something, but won't it just appear as:

SomeTensorListConstructor (A) -> TensorListResize (B) -> SomeTensorListReaderOrWriter (C)

In this case, C will search for its corresponding tensor list constructor (A) by passing straight through B. This will add A and C to the map, and they will get color matched. (I tried this out in a simple test).

If this is correct, then we can just add TensorListResize to supported_list_ops too and it should be good. Is there some way that it could go wrong?

I also notice that TensorListResize is marked as stateful, but I don't understand why?

Reply:
Ah you are right, TensorListResize can just be added to supported_list_ops. 

Good catch on TensorListResize being stateful. This is probably a bug. I'll ask the author.

Reply:
I confirmed TensorListResize being stateful is a bug. Feel free to create a PR fixing this.